django/forms/jinja2/django/forms/widgets/clearable_file_input.html

```diff
@@ -1,5 +1,5 @@
 {% if widget.is_initial %}{{ widget.initial_text }}: <a href="{{ widget.value.url }}">{{ widget.value }}</a>{% if not widget.required %}
-<input type="checkbox" name="{{ widget.checkbox_name }}" id="{{ widget.checkbox_id }}">
+<input type="checkbox" name="{{ widget.checkbox_name }}" id="{{ widget.checkbox_id }}" {% if widget.attrs.disabled %}disabled{% endif %}>
```

Comment:
How about `{% include "django/forms/widgets/attrs.html" %}` rather than only applying the disabled attribute? 

Seems to me like there could be a use case to try and pass in some css here. 

On the other hand folks may be doing this already to apply to the file `<input>` and applying these to the checkbox as well may be backwards incompatible.

Reply:
@smithdc1 thanks for the feedback

Using `{% include "django/forms/widgets/attrs.html" %}` also works (I've just executed the unit tests with this version). I have to admit that this was my first version but I wasn't sure about the backwards incompatibility potential problems.

Do you have any opinion on how to proceed?

Reply:
I had a look back at some history and can't see any previous commnets on the topic
Here is the code pre-widget templates [link v1.9](https://github.com/django/django/blob/e403f2217e67301526a0df5ea5be42e190e00e10/django/forms/widgets.py#L410)
Feels like what you have done is the safest option, and no one has raised a ticket to ask for the extra capability... so leave it as per your current proposal? 

The ticket also has `needs tests` still ticked - were you thinking of adding anything else?

Reply:
> I had a look back at some history and can't see any previous commnets on the topic
> Here is the code pre-widget templates [link v1.9](https://github.com/django/django/blob/e403f2217e67301526a0df5ea5be42e190e00e10/django/forms/widgets.py#L410)
> Feels like what you have done is the safest option, and no one has raised a ticket to ask for the extra capability... so leave it as per your current proposal?

I'm happy with the safe current proposal to fix the bug (checkbox enabled when the field is disabled)

I thought (but I wouldn't do right now in this PR, maybe in another one): if the 'attrs' could be expanded so it has `attrs.checkbox` (to be applied in the checkbox). For the `input type=file` it would ignore `attrs.checkbox`. Or some similar pattern to this.

> The ticket also has `needs tests` still ticked - were you thinking of adding anything else?

No, I didn't plan to add more. I've just unchecked it. The new test `test_clear_and_input_file_disabled_if_field_disabled` tests the new code (and actually it also testes the disabled in the input).

Thanks!

Reply:
@smithdc1 : is anything else that would be needed here? I haven't contributed to Django (well, a tiny thing years ago) so I'm unsure of the timings/steps (besides what I read on the documentation)

Thanks again!